OmegaExamples/src/probmods/generative-models.jl

```diff
@@ -0,0 +1,612 @@
+"""
+
+# ╔═╡ d70b97b5-af55-46f9-904e-309713603155
+md"""
+Conceptually, sampling from a random variable is done in the following two distinct steps:
+1. Construct a random $ω : Ω$, where $Ω$ can be though of as the space of exogenous variables as in causal graphical modelling nomenclature or simply the random number generator, `rng::Random.AbstractRNG`.
+2. Apply $ω$ to the random variable : $A(ω)$
+"""
+
+# ╔═╡ 0360ccef-ccf4-4284-80a9-ee5ed40becc4
+md"### Primitive and Composite Classes in Omega"
+
+# ╔═╡ 9998d466-eb74-476f-a768-d378bbe83f1b
+md"""##### Primitive Classes
+Omega comes with a set of built-in primitive random variable classes, such as `StdNormal` and `StdUniform`.  There are parameterless infinite sets of random variables.
+"""
+
+# ╔═╡ a0d64a28-e4f9-4003-b69d-7b27bd757c24
+As = StdUniform{Float64}()
+
+# ╔═╡ 84acb184-1c92-4761-a44e-bd1503f8255d
+rv = 1 ~ As
+
+# ╔═╡ 4e80ce71-6757-4908-8cd0-e382b11b70c3
+randsample(rv)
+
+# ╔═╡ 5fd22c50-4c81-41b2-a959-fe3dec00a6fd
+md"""
+##### Composite Class
+
+A class in Omega is actually just function of the form f(id, ω). Of course, you can specify your own classes simply by constructing a function.
+"""
+
+# ╔═╡ 8ac4cad4-eb4c-4e33-815c-e365668b3810
+μ = 2 ~ StdNormal{Float64}()
+
+# ╔═╡ 5fcf6c97-6f81-4be2-b27a-25b2abcabca1
+x(id, ω) = (id ~ Normal(μ(ω), 1))(ω)
+
+# ╔═╡ 1b4d9047-abb0-4fb7-ab72-0a4f790b1515
+x_ = 3 ~ x
+
+# ╔═╡ 0fe48b01-e908-4e6c-8c4c-b68cf8c172dc
+md"`x_` is a random variable of the class `x`"
+
+# ╔═╡ 0a2c3c83-8a67-4666-80c0-f580233dc019
+randsample((μ, x_))
+
+# ╔═╡ f2a20f84-b2e6-4e6c-ab51-6e7ea045954e
+md"Alternatively, `@joint` is used to create the joint distribution of variables:"
+
+# ╔═╡ 7d5587d1-c4b4-48f0-971c-677fbe268e73
+joint = @joint μ x_
+
+# ╔═╡ ae0ad1df-ab2a-4704-a121-623e05f079da
+randsample(joint)
+
+# ╔═╡ 0ae343fa-bc62-48ac-a34e-d21028531530
+md"""
+Every time we create a new (independent) random variable, we need to provide a unique id to it. We could instead use `@~` to generate `id`s automatically. For example, to simulate a (possibly biased) coin toss, we can simply write:
+"""
+
+# ╔═╡ 3c73f032-eedf-4d99-8489-a98ccf4b8805
+coin_toss = @~ Bernoulli()
+
+# ╔═╡ 61f4d77b-eecf-4cca-ac01-944efe58d9bb
+randsample(coin_toss)
+
+# ╔═╡ 2ad90265-8ef2-480f-81b0-7ed1a674011b
+md"If you run the program many times, and collect the values in a histogram, you can see what a typical sample looks like:"
+
+# ╔═╡ 9504162b-52e1-476d-8218-0afc2555e278
+viz(randsample(coin_toss, 1000))
+
+# ╔═╡ 15920581-2258-42e1-ba73-da82b2baa7e0
+md"As you can see, the result is an approximately uniform distribution over true(1) and false(0).
+This way we can construct more complex expressions that describe more complicated random variables. For instance, here we describe a process that samples a number adding up several independent Bernoulli distributions:"
+
+# ╔═╡ 3f04b226-a38a-4804-8691-59332bfc8728
+b_sum = (@~ Bernoulli()) +ₚ (@~ Bernoulli()) +ₚ (@~ Bernoulli())
```

Comment:
+ₚ is not defined for me:
 
<img width="948" alt="image" src="https://user-images.githubusercontent.com/895554/163176165-63f925b3-ae53-41e4-be1b-42a0d155ff6b.png">

I'm using Julia 1.7.0 and I followed the instructions @archanarw gave me.
<img width="844" alt="image" src="https://user-images.githubusercontent.com/895554/163176934-5c7da303-3c7f-4430-aef4-27f9dbafd656.png">

Please let me know how to fix it and then I'll continue with the PR.